src/fiatconnect/clients.ts

```diff
@@ -21,7 +21,8 @@ export function getSigningFunction(wallet: UnlockableWallet): (message: string)
 
 export async function getFiatConnectClient(
   providerId: string,
-  providerBaseUrl: string
+  providerBaseUrl: string,
+  providerApiKey?: string
 ): Promise<FiatConnectApiClient> {
   if (!fiatConnectClients[providerId] || fiatConnectClients[providerId].url !== providerBaseUrl) {
```

Comment:
I was planning to add an apiKey check here too, but I'm not sure that the second check actually does anything currently. Testing with manually changing the URL didn't update the baseUrl. I believe this is because getting new providers only occurs when the FiatExchange component mounts. It doesn't appear to be called again if you exit the add/withdraw flow and return to it later (or exit the app for that matter). 

I guess my question is when does the FiatExchange component unmount? I was only able to get it to unmount with a swipe up to fully close out the app.

Reply:
I think the intention of this check is to create a new client when (if) we update the `url` field in `valora-rest-api`. Probably good to also create a new client if apiKey changes too so I'd add a new check here for that as well

Reply:
Yeah I just added the check, but what I'm getting at is that I'm not sure that the code will ever hit the `url` (or for that matter `apiKey`) check. This is because fetching providers, via the `fetchFiatConnectProviders` function, only occurs when the FiatExchange component mounts. Any time that `getFiatConnectClient` is called, the wallet hasn't made a new request to `fetchFiatConnectProviders`. Instead, it just reuses the values it obtained from the first fetch. This means that providerBaseUrl and providerApiKey won't be different until FiatExchange un-mounts then re-mounts.

If `fetchFiatConnectProviders` were called as part of an error handling scenario, that'd be different I suppose.

Reply:
Hmm, I see what you're saying. I'd expect the FiatExchange component to unmount if you go to a different part of the app and then come back to the cico flow. does it not?

Reply:
Not when I was toying with the iOS simulator. It only called `fetchFiatConnectProviders` again after I swiped up to remove the app from running in the background. Even exiting the app and re-entering it did not cause `fetchFiatConnectProviders` to be re-called.

Reply:
its probably outside the scope of this PR but we should consider moving that to be called more often